### DISCUSSION TITLE
SUMO simulation runs slower than the real time

### DISCUSSION THREAD

**Discussion by tian96-pu** [cyberbotics/webots#4732] [Q&A]
I connected the webots and sumo simulation in the webots. However, I found the SUMO simulation runs slower than the real time. Does anyone have find this issue before? Also, I am wondering if the this issue could be solved by using a more powerful computer?

**Comment by omichel** [comment]
Yes, using a more powerful computer should resolve the issue.

**Reply by tian96-pu** [reply]
Hi Michel,

I deployed the webots and SUMO co-simulation on a desktop. 
The configuration of the desktop is 11th Gen Intel(R) Core(TM) i9-11900, 32GB RAM, GPU: RTX 3080.
The simulation world is not complex. We have one roundabout and about 30 SUMO vehicles. The webots vehicle is controlled by the Logitech G29 steering wheel. 
The simulation can be run at 23 Hz most time, however, the simulation can be stuck and run only at 5 Hz at some random points. I am not sure how to solve this problem. We want this platform to be run smoothly at around 20 Hz.

Thanks.

**Reply by omichel** [reply]
I believe the slowdown is caused by the introduction of new cars in the simulation. These cars are actually spawned by the [SUMO supervisor](https://github.com/cyberbotics/webots/tree/master/projects/default/controllers/sumo_supervisor). I believe that is possible to get rid of this problem by having all the cars already present in the environment (but very far away from the viewpoint) and to move them to their SUMO location when SUMO requests them instead of spawning them. In order to achieve this, you will have to modify the SUMO supervisor and probably the world file to include all the cars from the beginning of the simulation.
